.devcontainer/Dockerfile

```diff
@@ -31,10 +31,20 @@ RUN /tmp/set_up_stackwalker.sh && \
 COPY --chown=app:app ./webapp/package*.json /webapp-frontend-deps/
 RUN cd /webapp-frontend-deps/ && npm install
 
+# NOTE(relud): these dependencies are installed separately, relocated, and patched to reference the
```

Comment:
Interesting. I was thinking I would vendor them because they're pure-python, won't change, and it's an easy temporary fix using a model we've done at Mozilla before. It also allows us to adjust the code if we ever need to.

I vendored html5lib in Bleach here in case that's useful:

https://github.com/mozilla/bleach/tree/main/bleach/_vendor

Is there an advantage to shading over vendoring?

Reply:
i think i may have had a problem with the vendored library not being in the default python path for use in scripts like `bin/es_cli.py`. 

shading doesn't add any licensing concerns or requirements, but vendoring means we are distributing the packages and need to ensure we are respecting relevant license requirements.

vendoring obscures what version of the library is installed, which i didn't like.

Reply:
When I was researching how to go about with the Elasticsearch migration, I had implemented vendoring so I had two es and es-dsl libraries side by side. I renamed the old es/es-dsl libraries adding the version to the package name. I also added a shim library to switch between the two and account for some of the library differences. It worked with licensing requirements. I think that was 5 years ago or so.

Having said that, I'm game for shading. We can do monkeypatching if we need to adjust the library.

Reply:
> Having said that, I'm game for shading.

r?